src/auth/authService.ts

```diff
@@ -8,13 +8,33 @@ import {
 import * as Sentry from '@sentry/browser';
 import HttpStatusCode from 'http-status-typed';
 import i18n from 'i18next';
+import to from 'await-to-js';
 
 import pickProfileApiToken from './pickProfileApiToken';
 import createHttpPoller, { HttpPoller } from './http-poller';
+import retryPollingUntilSuccessful from './http-poller-with-promises';
 
 const origin = window.location.origin;
 export const API_TOKEN = 'apiToken';
 
+function isUserExpired(user?: Partial<User> | null): boolean {
+  if (!user) {
+    return true;
+  }
+  if (user.expired !== undefined) {
+    return user.expired;
+  }
+  const getExpiresIn = (expiresAtInSeconds?: number): number => {
+    if (expiresAtInSeconds) {
+      const now = Date.now() / 1000;
+      return expiresAtInSeconds - now;
+    }
+    return 0;
+  };
+
+  return getExpiresIn(user.expires_at) <= 0;
```

Comment:
I find this a bit hard to read. How about simply:

```suggestion
  if (user.expires_at) {
    return user.expires_at < Date.now() / 1000;
  }
  
  return true;
```

Reply:
Yeah, true. Previously the returned time was used in another context, but no need to return numbers anymore.

Reply:
Now it's a bit more easier to follow the logic. But there still is the helper function that is called from only one place. While the name of the function provides some documentation, I think in this case it's not really necessary. Comparing some timestamp to the current time is pretty basic stuff.

Reply:
Removed the inner helper function and added tests for the isAuthenticatedUser() which uses this function.